tools/check_format_test_helper.py

```diff
@@ -143,6 +143,9 @@ def checkFileExpectingOK(filename):
                                     "Don't use <mutex> or <condition_variable*>")
   errors += fixFileExpectingFailure("condition_variable_any.cc",
                                     "Don't use <mutex> or <condition_variable*>")
+  errors += checkFileExpectingError("shared_mutex.cc", "shared_mutex")
+  errors += fixFileExpectingFailure("shared_mutex.cc", "shared_mutex")
```

Comment:
I think you are saying here that you want to, for a given error, put the check & fix tests together.

I think I agree with you, this is better, because it's easier to maintain as the list grows.  Can you add a TODO to change the others?